apiserver/common/networkingcommon/networkconfigapi_test.go

```diff
@@ -441,6 +438,45 @@ func (s *networkConfigSuite) TestUpdateMachineLinkLayerOpBridgedDeviceMovesAddre
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *networkConfigSuite) TestUpdateMachineLinkLayerOpReprocessesDevices(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	hwAddr := "aa:bb:cc:dd:ee:ff"
+
+	s.expectMachine()
+	mExp := s.machine.EXPECT()
+	mExp.AllLinkLayerDevices().Return(nil, nil).Times(2)
+	mExp.AllAddresses().Return(nil, nil).Times(2)
+
+	// Expect the device addition to be attempted twice.
+	mExp.AddLinkLayerDeviceOps(
+		state.LinkLayerDeviceArgs{
+			Name:        "eth0",
+			Type:        "ethernet",
+			MACAddress:  hwAddr,
+			IsAutoStart: true,
+			IsUp:        true,
+		},
+	).Return([]txn.Op{{}, {}}, nil).Times(2)
+
+	op := s.NewUpdateMachineLinkLayerOp(s.machine, network.InterfaceInfos{
+		{
+			InterfaceName: "eth0",
+			InterfaceType: "ethernet",
+			MACAddress:    hwAddr,
+			Origin:        network.OriginMachine,
+		},
+	})
+
+	_, err := op.Build(0)
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
Shouldn't there be a non-zero error return? Or is the idea "something else caused conflict and the txn to be aborted, so we want to try again"

Reply:
Yes, we're representing a failure to apply rather than a failure to build.